docs/querying.md

```diff
@@ -2,10 +2,83 @@
 
 The query macro and where clause combine to give you full control over your query.
 
+## Where
+
+Where is using a QueryBuilder that allows you to chain where clauses together to build up a complete query.
+```crystal
+posts = Post.where(published: true, author_id: User.first.id)
+```
+
+It supports different operators:
+```crystal
+Post.where(:created_at, :gt, Time.now - 7.days)
+```
+
+Supported operators are :eq, :gteq, :lteq, :neq, :gt, :lt, :nlt, :ngt, :ltgt, :in, :nin, :like, :nlike
+
+
+This allows you create snippets of clauses that can be easily chained together similar to scopes in Rails.
+```crystal
+class Post < Granite::Base
+...
+  def live
+    where(published: true)
+  end
+
+  def draft
+    where(published: false)
+  end
+
+  def last_week
+    where(:created_at, :gt, Time.now - 7.days)
+  end
+end
+
+posts = Post.live.last_week
```

Comment:
wait, does this work?

Reply:
no, it doesn't. :-(  It returns the QueryBuilder, not this Object so it doesn't chain like I expected.  Is there anyway to get this to work?  Maybe with proc's?

Reply:
I will remove this example for now

Reply:
Rails suffers from the same problem, and I think they solve it by creating a separate scope builder which is capable of dereferencing further named scope calls. 

Even if you delegate back to the model (somehow) from the QueryBuilder, it'd have to pass the builder instance into the method to be able to chain. Both of these where's start from scratch with a fresh QueryBuilder:

```crystal
def named_scope_1
   where(column: value)
end

def named_scope_2
  where(column: value)
end
```

Potentially, if [instead of returning `self`](https://github.com/amberframework/granite/blob/master/src/granite/query/builder.cr#L61), each builder method returned a dup/clone, then I could see a possibility of something like this:

```crystal
macro named_scope(name)
  def {{name}}_scope_chain(context)
    yield context
  end
end

named_scope :last_week do |c|
  c.where(date: 1.week.ago)
end
```